src/main/java/uk/gov/pay/products/client/publicauth/model/CreateApiTokenRequest.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.pay.products.client.publicauth.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.pay.commons.model.TokenPaymentType;
+
+import javax.validation.constraints.NotNull;
+
+import static java.util.UUID.randomUUID;
+import static uk.gov.pay.commons.model.TokenPaymentType.CARD;
+import static uk.gov.pay.products.client.publicauth.model.TokenSource.PRODUCTS;
+
+public class CreateApiTokenRequest {
```

Comment:
I understand from comments on https://github.com/alphagov/pay-products/pull/570 that this was copied directly from public auth. I think we’re fine to let this diverge and represent the request as products sees it rather than as public auth sees it. For example, public auth’s idea of what’s mandatory and what to default to don’t really affect how products uses it.

(If we were using this request in many places then I could see the value of keeping the class consistent everywhere — and having it live in Pay Java commons — but right now I don’t think we’re using it enough to make the overhead worth it.)

Reply:
I have dropped TokenLink. It is not a mandatory field.